File: filosofos_java/src/EjecutarHilos.java
import hilos.Mesa;

/**
 * Script para ejecutar únicamente la solución con HILOS.
 * 
 * Uso:
 *     java EjecutarHilos [num_filosofos] [duracion_segundos]
 * 
 * Ejemplo:
 *     java EjecutarHilos 5 30
 */
public class EjecutarHilos {
    
    public static void main(String[] args) {
        // Parsear argumentos de línea de comandos
        int numFilosofos = 5;
        int duracion = 30;
        
        if (args.length > 0) {
            try {
                numFilosofos = Integer.parseInt(args[0]);
            } catch (NumberFormatException e) {
                System.err.println("Error: El número de filósofos debe ser un entero.");
                System.exit(1);
            }
        }
        
        if (args.length > 1) {
            try {
                duracion = Integer.parseInt(args[1]);
            } catch (NumberFormatException e) {
                System.err.println("Error: La duración debe ser un entero.");
                System.exit(1);
            }
        }
        
        System.out.println("\n" + "=".repeat(70));
        System.out.println("PROBLEMA DE LOS FILÓSOFOS COMENSALES - SOLUCIÓN CON HILOS");
        System.out.println("=".repeat(70));
        System.out.println("Configuración:");
        System.out.println("  - Número de filósofos: " + numFilosofos);
        System.out.println("  - Duración: " + duracion + " segundos");
        System.out.println("=".repeat(70) + "\n");
        
        try {
            // Crear la mesa
            Mesa mesa = new Mesa(numFilosofos);
            
            // Iniciar la simulación
            mesa.iniciarCena();
            
            System.out.println("\nSimulación corriendo por " + duracion + " segundos...");
            System.out.println("Presiona Ctrl+C para detener antes.\n");
            
            // Ejecutar por el tiempo especificado
            Thread.sleep(duracion * 1000L);
            
            System.out.println("\n" + "=".repeat(70));
            System.out.println("Fin de la simulación");
            System.out.println("=".repeat(70) + "\n");
            
        } catch (InterruptedException e) {
            System.out.println("\n\nSimulación interrumpida por el usuario.\n");
        }
        
        // Dar tiempo para que los hilos terminen sus operaciones actuales
        try {
            Thread.sleep(1000);
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
        
        System.exit(0);
    }
}

File: filosofos_python/ejecutar_hilos.py
"""
Script para ejecutar únicamente la solución con HILOS.

Uso:
    python ejecutar_hilos.py [num_filosofos] [duracion_segundos]

Ejemplo:
    python ejecutar_hilos.py 5 30
"""

import sys
import time
from solucion_hilos.mesa import Mesa


def main():
    # Parsear argumentos de línea de comandos
    num_filosofos = 5
    duracion = 30
    
    if len(sys.argv) > 1:
        try:
            num_filosofos = int(sys.argv[1])
        except ValueError:
            print("Error: El número de filósofos debe ser un entero.")
            sys.exit(1)
    
    if len(sys.argv) > 2:
        try:
            duracion = int(sys.argv[2])
        except ValueError:
            print("Error: La duración debe ser un entero.")
            sys.exit(1)
    
    print("\n" + "="*70)
    print("PROBLEMA DE LOS FILÓSOFOS COMENSALES - SOLUCIÓN CON HILOS")
    print("="*70)
    print(f"Configuración:")
    print(f"  - Número de filósofos: {num_filosofos}")
    print(f"  - Duración: {duracion} segundos")
    print("="*70 + "\n")
    
    try:
        # Crear la mesa
        mesa = Mesa(num_filosofos)
        
        # Iniciar la simulación
        mesa.iniciar_cena()
        
        print(f"\nSimulación corriendo por {duracion} segundos...")
        print("Presiona Ctrl+C para detener antes.\n")
        
        # Ejecutar por el tiempo especificado
        time.sleep(duracion)
        
        print("\n" + "="*70)
        print("Fin de la simulación")
        print("="*70 + "\n")
        
        # Detener todos los filósofos y esperar a que terminen
        mesa.detener()
        
        # Imprimir estadísticas finales
        mesa.imprimir_estadisticas()
        
    except KeyboardInterrupt:
        print("\n\nSimulación interrumpida por el usuario.\n")


if __name__ == "__main__":
    main()
